include/envoy/secret/secret.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Secret {
+
+/**
+ * An instance of the TlsCertificateSecret
```

Comment:
Following the other change, this should be part of `Envoy::Ssl` and not `Envoy::Secret` (nothing Secret-specific here).

Also, `s/TlsCertificateSecret/TlsCertificateConfig/g`.